drivers/SmartThings/zigbee-button/src/zigbee-multi-button/heiman/init.lua

```diff
@@ -81,6 +96,7 @@ local heiman_device_handler = {
   zigbee_handlers = {
     cluster = {
       [Scenes.ID] = {
+        [Scenes.server.commands.RecallScene.ID] = scenes_cluster_handler,
         [0x07] = scenes_cluster_handler
```

Comment:
the DTH has more clusters that it handles here. In your testing, should those be included?

Reply:
@liuyd96 As Green said, [In DTH code](https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/zigbee-scene-keypad.src/zigbee-scene-keypad.groovy#L64), other private cluster(0x0017 / 0xFE05) is using.

`	if (descMap?.clusterInt == 0x0017 || descMap?.clusterInt == 0xFE05 || descMap?.clusterInt == 0x0005) {`

You already checked these cluster is not used for ''HS6SSB-W-EF-3.0', but how about other HEIMAN scene panel such as 'HS6SSA-W-EF-3.0'/'E-SceneSwitch-EM-3.0'.
Maybe, Are these private cluster for GDKES or ORVIBO not for HEIMAN?

Reply:
Hi, @greens @inasail .
Thanks for your comments.

I think your questions are same that current EdgeDriver doesn't match with DTH so I replied together.

Now I only have HS6SSB-W-EF-3.0 device and current EdgeDriver only supports SceneSwitch-EM-3.0
 (SceneSwitch-EM-3.0 and  E-SceneSwitch-EM-3.0 are the same device, am I right?).
[in DTH code](https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/eb3cee1775bc55148813909aa9e891631de1e2e8/devicetypes/smartthings/zigbee-scene-keypad.src/zigbee-scene-keypad.groovy#L32C151-L32C171), HS6SSB-W-EF-3.0 and E-SceneSwitch-EM-3.0 have the same incluster/outcluster.

I thought the current EdgeDriver could be used directly after adding fingerprint.

But after adding HS6SSB-W-EF-3.0 fingerprint, I found the current driver is not directly compatible with HS6SSB-W-EF-3.0.

E.g. 
My HS6SSB-W-EF-3.0 doesn't have battery and HS6SSB-W-EF-3.0 only send out Scenes Cluster/RecallScene Command.

So I added device model check to avoid impacting E-SceneSwitch-EM-3.0.

Back to your questions.

> the DTH has more clusters that it handles here. In your testing, should those be included?

In my testing, HS6SSB-W-EF-3.0 only needs to process Scenes Cluster/RecallScene Command.

> You already checked these cluster is not used for ''HS6SSB-W-EF-3.0', but how about other HEIMAN scene panel such as 'HS6SSA-W-EF-3.0'/'E-SceneSwitch-EM-3.0'.
Maybe, Are these private cluster for GDKES or ORVIBO not for HEIMAN?

I understand what you are worried about. 
But current driver has not supported other HEIMAN Scene Panel and I only have HS6SSB-W-EF-3.0.
It's hard for me to consider other devices now.

Reply:
Hi, @greens @inasail .
Thanks for your comments.

I think your questions are same that current EdgeDriver doesn't match with DTH so I replied together.

Now I only have HS6SSB-W-EF-3.0 device. It doesn't have battery and
 HS6SSB-W-EF-3.0 only send out Scenes Cluster/RecallScene command.

So I added device model check to avoid impacting E-SceneSwitch-EM-3.0.

Back to your questions.

> the DTH has more clusters that it handles here. In your testing, should those be included?

In my testing, HS6SSB-W-EF-3.0 only needs to process Scenes Cluster/RecallScene Command.

> You already checked these cluster is not used for ''HS6SSB-W-EF-3.0', but how about other HEIMAN scene panel such as 'HS6SSA-W-EF-3.0'/'E-SceneSwitch-EM-3.0'.
Maybe, Are these private cluster for GDKES or ORVIBO not for HEIMAN?

I understand what you are worried about. 
But current driver has not supported other HEIMAN Scene Panel and I only have HS6SSB-W-EF-3.0.
It's hard for me to consider other devices now.

Reply:
Okay, GDKES or ORVIBO is not our interest. If you already check the functionality with HEIMAN device. 
I think It would be okay to consider only scene cluster.

Reply:
There's no indication of which cluster is for which device in the DTH, so I was just verifying.

Reply:
> There's no indication of which cluster is for which device in the DTH, so I was just verifying.

Here is the driver log when I push HS6SSB-W-EF-3.0. 
```
2023-08-09T05:19:49.937119880+00:00 INFO Zigbee Button  <ZigbeeDevice: 7bc48995-7e50-4f91-b9b8-c41387336520 [0x5B28] (HEIMAN Scene Panel)> received Zigbee message: < ZigbeeMessageRx || type: 0x01, < AddressHeader || src_addr: 0x5B28, src_endpoint: 0x01, dest_addr: 0x0010, dest_endpoint: 0xFF, profile: 0x0104, cluster: Scenes >, lqi: 0xB2, rssi: -64, body_length: 0x0008, < ZCLMessageBody || < ZCLHeader || frame_ctrl: 0x01, seqno: 0x17, ZCLCommandId: 0x05 >, < RecallScene || group_id: 0x0010, scene_id: 0x02, transition_time: 0x0001 > > >

2023-08-09T05:19:50.030261619+00:00 TRACE Zigbee Button  Found ZigbeeMessageDispatcher handler in zigbee_button -> ZigBee multi button -> Heiman Device handler

2023-08-09T05:19:50.031521323+00:00 INFO Zigbee Button  Executing ZclClusterCommandHandler: cluster: Scenes, command: RecallScene

2023-08-09T05:19:50.040452521+00:00 INFO Zigbee Button  <ZigbeeDevice: 7bc48995-7e50-4f91-b9b8-c41387336520 [0x5B28] (HEIMAN Scene Panel)> emitting event: {"capability_id":"button","component_id":"button1","state_change":true,"state":{"value":"pushed"},"attribute_id":"button"}

2023-08-09T05:19:50.054886634+00:00 INFO Zigbee Button  <ZigbeeDevice: 7bc48995-7e50-4f91-b9b8-c41387336520 [0x5B28] (HEIMAN Scene Panel)> emitting event: {"capability_id":"button","component_id":"main","state_change":true,"state":{"value":"pushed"},"attribute_id":"button"}

2023-08-09T05:19:50.070640420+00:00 DEBUG Zigbee Button  HEIMAN Scene Panel device thread event handled

2023-08-09T05:19:51.584658366+00:00 TRACE Zigbee Button  Received event with handler zigbee

2023-08-09T05:19:51.587393664+00:00 INFO Zigbee Button  <ZigbeeDevice: 7bc48995-7e50-4f91-b9b8-c41387336520 [0x5B28] (HEIMAN Scene Panel)> received Zigbee message: < ZigbeeMessageRx || type: 0x01, < AddressHeader || src_addr: 0x5B28, src_endpoint: 0x01, dest_addr: 0x0012, dest_endpoint: 0xFF, profile: 0x0104, cluster: Scenes >, lqi: 0xA4, rssi: -63, body_length: 0x0008, < ZCLMessageBody || < ZCLHeader || frame_ctrl: 0x01, seqno: 0x18, ZCLCommandId: 0x05 >, < RecallScene || group_id: 0x0012, scene_id: 0x04, transition_time: 0x0001 > > >

2023-08-09T05:19:51.630981193+00:00 TRACE Zigbee Button  Found ZigbeeMessageDispatcher handler in zigbee_button -> ZigBee multi button -> Heiman Device handler

2023-08-09T05:19:51.632225570+00:00 INFO Zigbee Button  Executing ZclClusterCommandHandler: cluster: Scenes, command: RecallScene

2023-08-09T05:19:51.640969196+00:00 INFO Zigbee Button  <ZigbeeDevice: 7bc48995-7e50-4f91-b9b8-c41387336520 [0x5B28] (HEIMAN Scene Panel)> emitting event: {"capability_id":"button","component_id":"button2","state_change":true,"state":{"value":"pushed"},"attribute_id":"button"}

2023-08-09T05:19:51.652898562+00:00 INFO Zigbee Button  <ZigbeeDevice: 7bc48995-7e50-4f91-b9b8-c41387336520 [0x5B28] (HEIMAN Scene Panel)> emitting event: {"capability_id":"button","component_id":"main","state_change":true,"state":{"value":"pushed"},"attribute_id":"button"}

2023-08-09T05:19:51.664320515+00:00 DEBUG Zigbee Button  HEIMAN Scene Panel device thread event handled

2023-08-09T05:19:53.592740157+00:00 TRACE Zigbee Button  Received event with handler zigbee

2023-08-09T05:19:53.594695179+00:00 INFO Zigbee Button  <ZigbeeDevice: 7bc48995-7e50-4f91-b9b8-c41387336520 [0x5B28] (HEIMAN Scene Panel)> received Zigbee message: < ZigbeeMessageRx || type: 0x01, < AddressHeader || src_addr: 0x5B28, src_endpoint: 0x01, dest_addr: 0x0011, dest_endpoint: 0xFF, profile: 0x0104, cluster: Scenes >, lqi: 0xC0, rssi: -66, body_length: 0x0008, < ZCLMessageBody || < ZCLHeader || frame_ctrl: 0x01, seqno: 0x19, ZCLCommandId: 0x05 >, < RecallScene || group_id: 0x0011, scene_id: 0x03, transition_time: 0x0001 > > >

2023-08-09T05:19:53.679515081+00:00 TRACE Zigbee Button  Found ZigbeeMessageDispatcher handler in zigbee_button -> ZigBee multi button -> Heiman Device handler

2023-08-09T05:19:53.680779448+00:00 INFO Zigbee Button  Executing ZclClusterCommandHandler: cluster: Scenes, command: RecallScene

2023-08-09T05:19:53.695247544+00:00 INFO Zigbee Button  <ZigbeeDevice: 7bc48995-7e50-4f91-b9b8-c41387336520 [0x5B28] (HEIMAN Scene Panel)> emitting event: {"capability_id":"button","component_id":"button3","state_change":true,"state":{"value":"pushed"},"attribute_id":"button"}

2023-08-09T05:19:53.714045807+00:00 INFO Zigbee Button  <ZigbeeDevice: 7bc48995-7e50-4f91-b9b8-c41387336520 [0x5B28] (HEIMAN Scene Panel)> emitting event: {"capability_id":"button","component_id":"main","state_change":true,"state":{"value":"pushed"},"attribute_id":"button"}

2023-08-09T05:19:53.725221549+00:00 DEBUG Zigbee Button  HEIMAN Scene Panel device thread event handled
```
Just for HS6SSB-W-EF-3.0, processing Scene Cluster is enough.

Thanks.